### PR TITLE
Search page

### PR DESCRIPTION
**Pull request by wejendorp**
**TODO**
Figure out weird Document error.

**DONE**
- Using a new json lib, and moved serialization into an AppModel class.
- CreateDocument logic for full-text search and search controller.

**Template**
Cleaner JS, but no results to display.

### CODE REVIEW

File: controllers/searchController.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from util import AppRequestHandler
+from models.study import Study, Snp, Gene, Disease
+
+class SNPSearch(AppRequestHandler):
+    """ Handle SNP searches """
+    def get(self, query):
+        """ Request via GET = single """
+        snp = None
+        if query != "":
+            snp = Snp.gql('WHERE snpid = :1', query)
```
Comment (jensraaby) [inline controllers/searchController.py]: There seems to be duplication between this method and the ones in the other xxxSearch classes  - could it be shared?

**Comment by jensraaby** [comment]
I like the search page in general, but if I type in plain text I get an alert and it doesn't search. Is that intended? If you can search for a disease it will probably be in plain text

**Comment by wejendorp** [comment]
Oh right, yea it is still a work in progress, I totally forgot that the old JS function is still used for form submit.

**Comment by wejendorp** [comment]
Can you try and test the Text search pages after populating the database and see if you get the same error? That is, template not found error for the sub-categories on http://localhost:8080/_ah/admin/search
Doing a rebase now.

**Comment by paddie** [comment]
Wow, you really need to rebase :)

I'm testing as well..

**Comment by paddie** [comment]
And yes, also getting the error.

**Comment by wejendorp** [comment]
Note to self: Try remove default template fallback from requesthandler.

**Comment by wejendorp** [comment]
I am an idiot and did the fix of search on the development branch.

**Pull request closed without merging**